packager/live_packager.cc

```diff
@@ -223,4 +284,83 @@ Status LivePackager::Package(const Segment& in, FullSegmentBuffer& out) {
   return packager.Run();
 }
 
+SegmentManager::SegmentManager() = default;
+
+uint64_t SegmentManager::OnSegmentWrite(const std::string& name,
+                                        const void* buffer,
+                                        uint64_t size,
+                                        FullSegmentBuffer& out) {
+  out.AppendData(reinterpret_cast<const uint8_t*>(buffer), size);
+  return size;
+}
+
+void SegmentManager::InitializeEncryption(const LiveConfig& config,
+                                          EncryptionParams& encryption_params) {
+  // TODO: encryption for fmp4 will be added later
+  if (config.protection_scheme == LiveConfig::EncryptionScheme::SAMPLE_AES) {
+    // Internally shaka maps this to an internal code for sample aes
+    //
+    // This is a fake protection scheme fourcc code to indicate Apple Sample
+    // AES. FOURCC_cbca = 0x63626361,
+    //
+    // Additionally this seems to be the recommended protection schema to when
+    // using the shaka CLI:
+    // https://shaka-project.github.io/shaka-packager/html/tutorials/raw_key.html
+    encryption_params.protection_scheme =
+        EncryptionParams::kProtectionSchemeCbcs;
+
+    encryption_params.key_provider = KeyProvider::kRawKey;
+    RawKeyParams::KeyInfo& key_info = encryption_params.raw_key.key_map[""];
+    key_info.key = config.key;
+    key_info.key_id = config.key_id;
+    key_info.iv = config.iv;
+  }
+}
+
+Aes128EncryptedSegmentManager::Aes128EncryptedSegmentManager(
+    const std::vector<uint8_t>& key,
+    const std::vector<uint8_t>& iv)
+    : SegmentManager(),
+      encryptor_(new media::AesCbcEncryptor(media::kPkcs5Padding,
+                                            media::AesCryptor::kUseConstantIv)),
+      key_(key),
+      iv_(iv) {}
+
+Aes128EncryptedSegmentManager::~Aes128EncryptedSegmentManager() = default;
+
+uint64_t Aes128EncryptedSegmentManager::OnSegmentWrite(const std::string& name,
+                                                       const void* buffer,
+                                                       uint64_t size,
+                                                       FullSegmentBuffer& out) {
+  if (!encryptor_->InitializeWithIv(key_, iv_)) {
```

Comment:
could we do this only once inside a constructor?

Reply:
I could see the challenge of moving this into the ctor due to `InitializeWithIv()` could fail and we would have to deal with that in the ctor.  We could throw and exception from the ctor, handle it, and terminate early by returning a non-OK status.  The alternative is to add an `Init()` function to this class and check its return status.  Shaka doesn't seem to throw exceptions, so the latter may be more consistent with the codebase.

Reply:
Oh, looks like we can use move this to `InitializeEncryption()` and change the return type.